### PR TITLE
ui: Fixes disposing on GTK/Avalonia and Firmware Messages on Avalonia

### PR DESCRIPTION
**Pull request by AcK77**
This fixes a dispose issue under Horizon/GTK/Avalonia, we don't check if the ApplicationClient is null so it throw NRE. We don't check if the main loop is active and waiting an event which is set in the main loop... So that could lead to a freeze.
Avalonia code is running on a main thread and waiting for the popup result, popup are running on a UI thread for the main thread never get the result and freeze.

Everything works fine in GTK/Avalonia (Thanks to @TSRBerry and @emmauss for their help) now.

Fixes https://github.com/Ryujinx/Ryujinx/issues/3873

### CODE REVIEW

File: Ryujinx/Ui/VKRenderer.cs [hunk 1]
```diff
@@ -72,7 +72,11 @@ protected override string GetGpuBackendName()
 
         protected override void Dispose(bool disposing)
         {
-            Device.DisposeGpu();
+            if (Device != null)
+            {
+                Device.DisposeGpu();
+            }
```
Comment (TSRBerry) [inline Ryujinx/Ui/VKRenderer.cs:78]:
```suggestion
                Device?.DisposeGpu();
```
Could this be written like that?

Comment (AcK77) [reply Ryujinx/Ui/VKRenderer.cs:78]: True!

**Comment by TSRBerry** [comment]
This should fix everything on the Avalonia side. I tested this on Linux.

**Comment by AcK77** [comment]
Still freeze on windows for me using Avalonia :(

**Comment by AcK77** [comment]
Everything should works as expected now.

**Review by marysaka — APPROVED** [review]

**Review by emmauss — APPROVED** [review]

**Pull request merged**